java-util/src/main/java/io/druid/java/util/common/io/smoosh/FileSmoosher.java

```diff
@@ -347,17 +367,21 @@ public boolean isOpen()
   public void close() throws IOException
   {
     //book keeping checks on created file.
-    if (!completedFiles.isEmpty() || !filesInProcess.isEmpty())
-    {
-      for (File file: completedFiles)
-      {
-        file.delete();
+    if (!completedFiles.isEmpty() || !filesInProcess.isEmpty()) {
+      for (File file : completedFiles) {
+        if (!file.delete()) {
+          LOG.warn("Unable to delete file [%s]", file.getPath());
+        }
       }
-      for (File file: filesInProcess)
-      {
-        file.delete();
+      for (File file : filesInProcess) {
+        if (!file.delete()) {
+          LOG.warn("Unable to delete file [%s]", file.getPath());
+        }
       }
-      throw new ISE(String.format("%d writers needs to be closed before closing smoosher.", filesInProcess.size() + completedFiles.size()));
+      throw new ISE(String.format(
```

Comment:
`ISE` does a `String.format` for you, don't need to duplicate it here.  Also, the message is probably a lot nicer if you expose both numbers instead of only exposing the sum.